craft_parts/executor/step_handler.py

```diff
@@ -267,7 +264,43 @@ def _handle_control_api(self, scriptlet_name, function_call) -> None:
                 )
 
         function_name = function_json["function"]
+        function_args = function_json["args"]
 
+        invalid_control_api_call = functools.partial(
+            errors.InvalidControlAPICall,
+            part_name=self._part.name,
+            scriptlet_name=scriptlet_name,
+        )
+
+        if function_name in ["pull", "build", "stage", "prime"]:
+            if len(function_args) > 0:
+                raise invalid_control_api_call(
+                    message=f"invalid arguments to function {function_name!r}",
+                )
+            self._handle_step_function(function_name)
+        elif function_name == "set":
+            if len(function_args) != 2:
+                raise invalid_control_api_call(
+                    message=(
+                        f"invalid number of arguments to function {function_name!r}"
+                    ),
+                )
+            name = function_args[0]
+            value = function_args[1]
```

Comment:
you already know that length is 2, so much simpler and readable to do `name, value = function_args`